### DISCUSSION TITLE
Ultrasonic sensor JSN-SR04T (ESP 8266 Node mcu) - help needed

### DISCUSSION THREAD

**Discussion by jkmqtt** [arendst/Tasmota#15948] [Q&A] [answered]
i am making water level indicator . ( 5 level to  indicate ) using 5 led 

how to connect 5 led to indicate level 

how to put condition to 

LED 1 glow 💡 ( when distance = 20 cm )
LED 2 glow 💡 ( when distance = 40 cm )
LED 3 glow 💡 ( when distance = 60 cm )
LED 4 glow 💡 ( when distance = 80 cm )
LED 5 glow 💡 ( when distance = 100 cm )
Motor stop 🛑 ( when distance = 100 cm )

HOW  to interface gpio pin high/low to glow led using JSN-SR04T sensor distance.

**Comment by barbudor** [comment, marked as answer]
As you want to control 5 Leds + 1 motor, I suggest to use "Relay" for all because you would otherwise face some limitations
So assuming LED x is connected to GPIO Relay X and the motor is connected to GPIO Relay 6

The SENSOR payload for the SR04 is
```
{"Time":"2019-01-01T22:42:35","SR04":{"Distance":16.754}}
```
So your main rule trigger will be `SR04#Distance`
It's a good opportunity to use the BREAK keyword in rule : https://tasmota.github.io/docs/Rules/#using-break-to-simulate-ifelseifelseendif

```
Rule1
  ON system#boot DO Power0 OFF ENDON 
  ON SR04#Distance>=100 DO Backlog Power1 1; Power2 1; Power3 1; Power4 1; Power5 1; Power6 0 BREAK
  ON SR04#Distance>=80  DO Backlog Power1 1; Power2 1; Power3 1; Power4 1; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=60  DO Backlog Power1 1; Power2 1; Power3 1; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=40  DO Backlog Power1 1; Power2 1; Power3 0; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=20  DO Backlog Power1 1; Power2 0; Power3 0; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance<20   DO Backlog Power1 0; Power2 0; Power3 0; Power4 0; Power5 0; Power6 1 ENDON
```
The order of the rule is important as the BREAK is (in that case) similar to an ELSE IF

If you want the rule to be executed only at teleperiod, replace the trigger by `Tele-SR04#Distance`

**Reply by jkmqtt** [reply]
thank you so much Sir 
for instant help 

------------------------------------------------------------------
This feature is not included in precompiled binaries.

To use it you must compile your build. Add the following to user_config_override.h:
------------------------------------------------------------------

to add this rule 
can i directly add these code to 

**user_config_override.h**


-----------------------------------------------------------------
```
{"Time":"2019-01-01T22:42:35","SR04":{"Distance":16.754}}

Rule1
  ON system#boot DO Power0 OFF ENDON 
  ON SR04#Distance>=100 DO Backlog Power1 1; Power2 1; Power3 1; Power4 1; Power5 1; Power6 0 BREAK
  ON SR04#Distance>=80  DO Backlog Power1 1; Power2 1; Power3 1; Power4 1; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=60  DO Backlog Power1 1; Power2 1; Power3 1; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=40  DO Backlog Power1 1; Power2 1; Power3 0; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance>=20  DO Backlog Power1 1; Power2 0; Power3 0; Power4 0; Power5 0; Power6 1 BREAK
  ON SR04#Distance<20   DO Backlog Power1 0; Power2 0; Power3 0; Power4 0; Power5 0; Power6 1 ENDON
```
----------------------------------------------------------------------
please guide or give me direction to 
enable sensor and to add this rule.

**user_config_override.h** 

?

Thank you sir

**Reply by barbudor** [reply]
I don't get your question
SR04 is included by default in tasmota-sensors, so you don't need to compile anything
As well as Rules

You need to properly configure Tasmota once flashed which includes entering rules from the console

Please read the doc : https://tasmota.github.io/docs/Rules/

**Reply by jkmqtt** [reply]
thank you so much.. 

i got my answer from your help 

i just to enable SR04  {if not enabled on flash  file } in **user_config_override.h**

then enter rule from console .
